processing/src/main/java/io/druid/segment/filter/LikeFilter.java

```diff
@@ -140,4 +139,60 @@ public boolean supportsBitmapIndex(BitmapIndexSelector selector)
   {
     return selector.getBitmapIndex(dimension) != null;
   }
+
+  private static Iterable<ImmutableBitmap> getBitmapIterator(
+      final BitmapIndex bitmapIndex,
+      final LikeDimFilter.LikeMatcher likeMatcher,
+      final Indexed<String> dimValues
+  )
+  {
+    final String lower = Strings.nullToEmpty(likeMatcher.getPrefix());
+    final String upper = Strings.nullToEmpty(likeMatcher.getPrefix()) + Character.MAX_VALUE;
+    final int startIndex; // inclusive
+    final int endIndex; // exclusive
+
+    final int lowerFound = bitmapIndex.getIndex(lower);
+    startIndex = lowerFound >= 0 ? lowerFound : -(lowerFound + 1);
+
+    final int upperFound = bitmapIndex.getIndex(upper);
+    endIndex = upperFound >= 0 ? upperFound + 1 : -(upperFound + 1);
+
+    return new Iterable<ImmutableBitmap>()
+    {
+      @Override
+      public Iterator<ImmutableBitmap> iterator()
+      {
+        return new Iterator<ImmutableBitmap>()
```

Comment:
Could you please apply here the same improvement? https://github.com/druid-io/druid/pull/3754/files